enhancements/sig-architecture/14-addon-cluster-proxy/README.md

```diff
@@ -0,0 +1,362 @@
+under a new group of api `proxy.addon.open-cluster-management.io/v1alpha1`:
+
+- __ManagedProxyServer__: the configuration and status of proxy-servers.
+
+- __MangedProxyAgent__: the configuration and status of proxy-agents.
+
+Additionally, the operator will be list-watching the standard `ManagedClusterAddon`
+resource and automatically installs proxy-agents w/ the expected options
+upon discovering the addon in each cluster's namespace.
+
+
+##### ManagedProxyServer
+
+The `ManagedProxyServer` is supposed to be a cluster-scoped resource and holds 
+information for provisioning an active proxy-server instance. An example will 
+be:
+
+```yaml
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ManagedProxyServer
+metadata:
+  name: default # Or any unique name, will be the name of the workload
+spec:
+  replicas: 1
+  certificates:
+    signingSANs: 
+      - "1.1.1.1"
+      - "example.com"
+    rotation:
+      expiringDays: 365
+      forceReloadDays: 200
+  proxy: 
+    mode: [http-connect|grpc|uds] 
+  deploy:
+    namespace: open-cluster-management
+    agentAuthentication:
+      namespace: <validating namespace>
+      serviceAccount: <validating service-account>
+    listeningPorts:
+      proxyServer: 8090
+      agentServer: 8091 
+      healthServer: 8092
+      adminServer: 8095 
+status:
+  lastObserverdGeneration: 2
+  conditions:
+  - type: "AllHealthy" 
+    status: "True"
+  - type: "AllUpToDate" 
+    status: "True"
+  components:
+  - name: default-1
+    namespace: open-cluster-management
+    healthy: true
+    restartCount: 0
+  - name: default-2
+    namespace: open-cluster-management
+    healthy: true
+    restartCount: 0
+```
+
+The operator will be actively syncing up the deployment of the proxy-servers
+to the configuration in the following steps:
+
+1. Signing all the required X509 certificates and keys then stores them into
+   a secret object:
+   
+    i.   a CSR requesting server certificate which will be shared across the 
+         replicas of the proxy servers.
+   
+    ii. a CSR requesting client certificate for logging onto in the tunnel 
+        (skipped if the proxy server under "uds" mode).
+
+2. Rolling/scaling out the proxy servers upon spec changes.
+   
+3. Actively probing/reporting the status of the proxy server instances.
+
+##### ManagedProxyAgent
+
+The `ManagedProxyAgent` resource is also a cluster-scoped resource and this
+is the template configuration for installing agents to specific clusters:
+
+An example is shown below:
+
+```yaml
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ManagedProxyAgent
+metadata:
+  name: default # Or any unique name, will be the name of the workload
+spec:
+  serverReference: 
+    # A pointer to the server configuration so that auto-signed certificates
+    # and other related config can be imported.
+    apiVersion: "addon.open-cluster-management.io/v1alpha1"
+    resource: managedproxyservers
```

Comment:
Do we expect there will be other resource option here?

Reply:
no actually, i was just following the prior art from rbac's role and rolebinding